src/super_gradients/training/utils/ema.py

```diff
@@ -30,7 +30,7 @@ class ModelEMA:
     GPU assignment and distributed training wrappers.
     """
 
-    def __init__(self, model, decay: float = 0.9999, beta: float = 15, exp_activation: bool = True):
+    def __init__(self, model, decay: float, decay_function: IDecayFunction):
```

Comment:
I think the proper way to handle this new signature is by using the `deprecate`  library (Important: note that this is "deprecate" without a d at the end. there are two python modules with close names).
you can see an example in the dataset_utils.

to use it, you will add a decorator on the __init__ function:
```
@deprecated(target=new_init, deprecated_in="3.0.6", remove_in="4.0.0", args_mapping=...,  args_extra=...)
def __init__(...)
```
in `args_mapping` and `args_extra` you can resolve all of the cases of new vs old arguments.
this library issues a formatted message with all of the information the user will need. 

we may need to figure out exactly how to use it in this unique case. since we don't want the user to start using another init function (we just want him to change the arguments). so perhaps instead of using the decorator, we can use `_raise_warn_arguments()` that issues the message, without defining a new target function

Reply:
If we don't use `@deprecated` decorator, than what's the point of using `deprecate` at all in this case? 
Side note: This library has as few as 46 starts on GitHub. This does not necessary means it is bad library, rather than it's support time can be questioned.